crates/hyperqueue/src/transfer/connection.rs

```diff
@@ -134,6 +134,28 @@ impl ClientSession {
     }
 }
 
+async fn try_connect_to_server(record: &AccessRecord) -> crate::Result<TcpStream> {
+    let address = format!("{}:{}", record.host(), record.server_port());
+    match TcpStream::connect(&address).await {
+        Ok(conn) => Ok(conn),
+        // 113 = EHOSTUNREACH on Linux. Replace with ErrorKind::HostUnreachable once it's stabilized
+        Err(error) if matches!(error.raw_os_error(), Some(113)) => {
+            let hostname = get_hostname(None);
+            if hostname == record.host() {
+                let address = format!("localhost:{}", record.server_port());
+                log::warn!(
+                    "Could not reach {address}. It's host matches the local hostname, retrying\
+with `address`.",
```

Comment:
I do not understand the warning message. Do you mean retrying with ``localhost``?
(Btw: If it is common on some systems, maybe it should be just debug message to not spam a user that cannot do anything about it)

Reply:
Oh, this should have been a format placeholder of course, fixed, thanks.
It only appears once per each client session, but you're right that it could be a bit annoying. Changed to debug log.